src/field/field.cxx

```diff
@@ -62,5 +61,6 @@ void Field::error(const char *s, ...) const {
   }
   
   throw BoutException("Error in field: %s", err_buffer);
+  delete[] err_buffer;
```

Comment:
This leads to a memory leak, no matter if caught or not. (not caught however doesn't really matter)
b) use std::string